tests/purs/warning/4268.purs

```diff
@@ -0,0 +1,4 @@
+module Main where
+
+f :: Partial => Int -> Int
+f 0 = f 1
```

Comment:
It may be a little counterintuitive, but this is the way we write a test that ensures that a warning isn't emitted. Tests in `passing` may or may not emit warnings, but a test in `warning` must emit exactly the number of warnings indicated by the comments. Other examples of this pattern:

```
tests/purs/warning/2383.purs
tests/purs/warning/CoercibleUnusedImport.purs
tests/purs/warning/CoercibleUnusedNewtypeCtorExplicitImport.purs
tests/purs/warning/CoercibleUnusedNewtypeCtorImplicitImport.purs
tests/purs/warning/KindReExport.purs
```

Reply:
Ah... Ok. Then this is good to go.

Reply:
I'm opening an issue to document this since I didn't know it myself.